### PR TITLE
feat: add StringSequenceToEmbedding transformer and layer

### PR DESCRIPTION
**Pull request by mruiyangyou**
## Summary

Adds a new `StringSequenceToEmbedding` transformer/layer pair that parses a delimited string of pre-computed embedding vectors into a dense `(seq_len, embedding_dim)` float tensor.

- Spark transformer (`StringSequenceToEmbeddingTransformer`) and Keras layer (`StringSequenceToEmbeddingLayer`) with parity behaviour.
- Input format: vectors separated by `sequence_separator` (default `,`), floats within a vector by `separator` (default `|`). Example: `"1|2|3,4|5|6"` with `seq_len=2, embedding_dim=3` → `[[1,2,3],[4,5,6]]`.
- Pads short sequences with `pad_value` (default `"0"`); truncates long ones.
- Optional `reverse` mode reverses only the non-pad portion of each sequence (useful for chronological → recency-first ordering).
- Layer drops a trailing size-1 input axis to match the `StringToStringListLayer` convention, so `(None, 1, 1)` inputs produce `(None, 1, seq_len, embedding_dim)` outputs without a downstream squeeze.
- Empty tokens (from leading/trailing/repeated separators or fully empty cells) are replaced with `pad_value` before `tf.strings.to_number` to avoid `StringToNumberOp` failures at graph execution.
- README updated with a row in the supported layers table.

## Test plan

- [x] `tests/kamae/tensorflow/layers/test_string_sequence_to_embedding.py` — 9 unit tests covering default/custom separators, padding, truncation, reverse, trailing-1 squeeze behaviour, empty/malformed inputs, config round-trip, and invalid args.
- [x] `tests/kamae/spark/transformers/test_string_sequence_to_embedding.py` — 6 tests including Spark/TF parity across separators, padding, and reverse modes.
- [x] `tests/kamae/tensorflow/test_layer_serialisation.py` — added the new layer to the serialisation matrix; passes `test_all_layers_tested_for_serialisation`.
- [x] No regressions in existing `string_to_string_list` tests.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

**Comment by georyetti** [comment]
Hey @mruiyangyou - this looks good at a glance, but we are gearing up to do a big new kamae v3 release in the next few weeks or so. Can this wait till after these? I can aid migrating this to keras 3.

File: src/kamae/keras/tensorflow/layers/string_sequence_to_embedding.py [hunk 1]
```diff
@@ -0,0 +1,187 @@
...
+        # does not fail on the empty string.
+        dense = tf.where(tf.equal(dense, ""), self.pad_value, dense)
+
+        floats = tf.strings.to_number(dense, out_type=tf.float32)
+        result = tf.reshape(floats, [-1, self.seq_len, self.embedding_dim])
+
+        if self.reverse:
+            # A row is considered padding iff all of its components are 0.
+            row_norms = tf.reduce_sum(tf.abs(result), axis=-1)
+            seq_lengths = tf.reduce_sum(tf.cast(row_norms > 0, tf.int32), axis=-1)
```
Comment (georyetti) [inline src/kamae/keras/tensorflow/layers/string_sequence_to_embedding.py]: pad value is configurable, so this only works if pad value is 0 no? If pad value is -1, then row_norms > 0 and its considered non padded?

File: src/kamae/spark/transformers/string_sequence_to_embedding.py [hunk 2]
```diff
@@ -0,0 +1,300 @@
...
+
+            # Count the number of non-pad vectors (a vector is pad iff all
+            # of its components are zero). Reverse only that prefix.
+            abs_sums = F.transform(
+                vectors,
+                lambda v: F.aggregate(
+                    v,
+                    F.lit(0.0),
+                    lambda acc, value: acc + F.abs(value),
+                ),
```
Comment (georyetti) [inline src/kamae/spark/transformers/string_sequence_to_embedding.py]: Same as above, pad value = -1 breaks this I think

File: tests/kamae/keras/tensorflow/layers/test_string_sequence_to_embedding.py [hunk 3]
```diff
@@ -0,0 +1,201 @@
...
+        output = layer(inputs)
+        assert output.shape == (2, 3, 2)
+        tf.debugging.assert_near(expected, output)
+
+    def test_reverse_reverses_only_non_pad_portion(self):
+        layer = StringSequenceToEmbeddingLayer(
+            name="reverse",
+            seq_len=4,
+            embedding_dim=2,
+            reverse=True,
```
Comment (georyetti) [inline tests/kamae/keras/tensorflow/layers/test_string_sequence_to_embedding.py:110]: Can we add a test for a non zero pad value

**Review by georyetti — CHANGES_REQUESTED** [review]
One main comment on padding value being assumed to be zero.

**Comment by georyetti** [comment]
Also since pad value is a string, what happens if the user specific pad_value = "hello"? We should validate that its a numeric string on setting in Spark and in constructor of keras layer

**Review by georyetti — APPROVED** [review]

**Review by mandrecki — APPROVED** [review]
